src/Chainweb/BlockHeader.hs

```diff
@@ -174,6 +174,16 @@ instance IsMerkleLogEntry ChainwebHashTag BlockHeight where
     {-# INLINE toMerkleNode #-}
     {-# INLINE fromMerkleNode #-}
 
+instance Semigroup BlockHeight where
+    BlockHeight a <> BlockHeight b = BlockHeight $ a + b
+    {-# INLINE (<>) #-}
+
+instance Monoid BlockHeight where
+    mempty = 0
+
+    mappend = (<>)
+    {-# INLINE mappend #-}
```

Comment:
Potentially dishonest to force the `Semigroup` instance to be additive, but Lens requires a `Monoid`, and we don't have `deriving via` available to us yet.

Reply:
Normally, one would use a `newtype` wrapper in cases like this. In this case an additive Semigroup seems fine. It measure the amount of work that went into a block. Addition seems the natural operation for this. Alternatives might be `Min` or `Max`, but definitely not multiplication. I think, addition is fine.